### PR TITLE
Added "reset arenaworld" and "reset honorworld" commands

### PR DESCRIPTION
**Pull request by Helias**
Added "reset arenaworld" and "reset honorworld" commands.

I took this commands from here: https://github.com/WarHead/uwom-server/blob/691b62a0054324e1e57405ff827c4dc6393d3bec/src/server/scripts/Commands/cs_reset.cpp
and I modified and translated it

### CODE REVIEW

**Comment by DDuarte** [comment]
A help text should be added to the `commands` table with the syntax and description of the commands.

File: src/server/scripts/Commands/cs_reset.cpp [hunk 1]
```diff
@@ -56,6 +58,45 @@ class reset_commandscript : public CommandScript
...
 
+    static bool HandleResetArenaWorldCommand(ChatHandler * handler, const char * args)
+    {
+
+        SQLTransaction trans = CharacterDatabase.BeginTransaction();
+
+        trans->PAppend("UPDATE `characters` SET `arenaPoints`=%u", sWorld->getIntConfig(CONFIG_START_ARENA_POINTS));
+
+        std::string param = args;
+        if (*args && param == "delete")
```
Comment (DDuarte) [inline src/server/scripts/Commands/cs_reset.cpp]: http://ideone.com/HIekfq

**Comment by Helias** [comment]
Here the comments about the commands
```SQL
DELETE FROM `command` WHERE `name`='reset arenaworld';
DELETE FROM `command` WHERE `name`='reset honorworld';

INSERT INTO `command` (`name`, `permission`, `help`) VALUES
('reset arenaworld', 50, 'Syntax: .reset arenaworld [delete]\nReset all arena points/rating at the default values. With [delete] all teams will be deleted.'),
('reset honorworld', 50, 'Syntax: .reset honorworld [kills]\nReset all honor points at the default values. With [kills] all the kills will be resetted.');
```
I don't know what should I put in "permission", anyway when I import this sql code, in the list of ".reset" commands I can't see my two commands that I imported

**Comment by Nyeriah** [comment]
They should use each their own permission (like the rest of the commands). So the permission to be used is the one you assign to them.

File: src/server/scripts/Commands/cs_reset.cpp [hunk 2]
```diff
@@ -56,6 +58,45 @@ class reset_commandscript : public CommandScript
         return commandTable;
     }
 
+    static bool HandleResetArenaWorldCommand(ChatHandler * handler, const char * args)
```
Comment (tkrokli) [inline src/server/scripts/Commands/cs_reset.cpp]:
```
/home/travis/build/TrinityCore/TrinityCore/src/server/scripts/Commands/cs_reset.cpp:61:60: fatal error: 
      unused parameter 'handler' [-Wunused-parameter]
    static bool HandleResetArenaWorldCommand(ChatHandler * handler, cons...
                                                           ^

**Comment by Shauren** [comment]
Result of this command will be overwritten for any online player when they save to db.

**Comment by Helias** [comment]
```Result of this command will be overwritten for any online player when they save to db.```

Indeed the author of this command added these to logout the players but after run the comand with these sort of world logout the players can't connect to the server so I decided to remove these part of code:

```C++
const SessionMap SMap = sWorld->GetAllSessions();
        for (SessionMap::const_iterator itr = SMap.begin(); itr != SMap.end(); ++itr)
            if (Player * plr = (*itr).second->GetPlayer())
            {
                if (plr->GetSession() != handler->GetSession()) // Den Ausführenden nicht ausloggen, da es sonst zu einem Crash kommt!!!
                    plr->GetSession()->LogoutPlayer(true); // Alle online Spieler ausloggen und speichern, damit die Änderungen sofort aktiv sind!
                else
                {
                    plr->SetArenaPoints(0);
                    plr->GetSession()->LogoutRequest(time(NULL));
                }
            }
```

Anyway .reset arenaworld should be used after closing/opening the arena seasons and before a restart of the server, normally the devleoper make it manually, so they execute manually a similar SQL code to this:

``UPDATE `arena_team` SET `rating`=%u,`seasonGames`=0,`seasonWins`=0,`weekGames`=0,`weekWins`=0,`rank`=0", sWorld->getIntConfig(CONFIG_ARENA_START_RATING));
UPDATE `arena_team_member` SET `personalRating`=%u,`weekGames`=0,`weekWins`=0,`seasonGames`=0,`seasonWins`=0", sWorld->getIntConfig(CONFIG_ARENA_START_PERSONAL_RATING));
UPDATE `character_arena_stats` SET `matchMakerRating`=%u", sWorld->getIntConfig(CONFIG_ARENA_START_MATCHMAKER_RATING));```

This command is only an easy way to do that with the core

**Comment by Helias** [comment]
I re-opened this pull request after testing and modifies here:
https://github.com/TrinityCore/TrinityCore/pull/15660

**Pull request closed without merging**